ALUM/ALUM/Components/ProgressBarComponent.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  ProgressBarComponent.swift
+//  ALUM
+//
+//  Created by Yash Ravipati on 1/25/23.
+//
+
+// nodeList - [completed, not completed, completed...]
+
+import SwiftUI
+
+struct ProgressBarComponent: View {
+    @State var nodes = 5
+    @State var filledNodes = 4
+    @State var activeNode = 2
+
+    var body: some View {
+        GeometryReader { (geometry) in
+            self.makeView(geometry)
+        }
+    }
+
+    func makeView(_ geometry: GeometryProxy) -> some View {
+        let margin = geometry.size.width / CGFloat(self.nodes) - 16
```

Comment:
During dev hours, we discussed that there was some extra width on the side but turns out that's just the padding from the circle. This padding is evenly distributed to both sides of the circle so that is how we can deterministically find the padding if we have screen width and the number of nodes.